src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,231 @@
+        }
+        String[] elements = keywords[1].split(" /by ", 2);
+        if (elements.length == 1) {
+            throw new DukeException("you need to tell me the time in format: (task) /by (time)!");
+        }
+        String time = elements[1];
+        String textDescription = elements[0];
+        Task deadline = new Deadline(textDescription, time);
+        listOfItems.addTask(deadline);
+        ui.printAdditionConfirmation(deadline, listOfItems);
+    }
+
+    /**
+     * Deletes an item from the list by passing the index of the task to be deleted.
+     *
+     * @param input Command line input from the user to be parsed.
+     * @throws DukeException Throws an exception if index is not on the list or with wrong input.
+     */
+    private void deleteItem(String input) throws DukeException {
+        try {
+            Integer idx = Integer.parseInt(parser.getDescription(input));
+            Task item = listOfItems.getTask(idx);
+            listOfItems.removeTask(idx);
+            ui.printDeletionConfirmation(item, listOfItems);
+        } catch (NumberFormatException e) {
+            throw new DukeException("you need to choose a number from the list in the form: delete (list index)!");
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("you need to pick an item from within the list!");
+        }
+    }
+
+    /**
+     * Displays a list of items on the command line that contain the keyword entered by the user.
+     *
+     * @param input Command line input from the user.
+     * @throws DukeException Throws an exception if no keyword was entered or if no tasks match the keyword given.
+     */
+    private void findItems(String input) throws DukeException {
+        try {
+            String keyword = parser.getDescription(input);
+            Tasklist validItems = listOfItems.findAllBy(keyword);
+            ui.outputWrapper(validItems);
+        }catch (ArrayIndexOutOfBoundsException e){
+            throw new DukeException("please input a keyword in the format: [find] (keyword)");
+        } catch (NullPointerException e) {
+            throw new DukeException("no tasks match your given keyword!");
+        }
+
+    }
+    /**
+     * Handles invalid commands from the user.
+     *
+     * @throws DukeException Throws an exception when a wrong input is given on the command line.
+     */
+    private void markAsInvalid(String input) throws DukeException {
+        if (input.equals("")) {
+            throw new DukeException("you need to type something!");
+        } else {
+            throw new DukeException("your command is invalid!");
+        }
+    }
+
+    /**
+     * Handles the user's input and determines which command should be run.
+     *
+     * @param input Command line input from the user.
+     */
+    private void handleInput(String input) {
+        String command = parser.getCommand(input);
+        try {
+            switch(command) {
```

Comment:
According to Java Coding standards, for a switch statement, case clauses aren't to be intended. I blame IntelliJ for this. You could try to fix this manually.

Reply:
yep i fixed the indentations